src/Dropdown/Dropdown.tsx

```diff
@@ -1,400 +1,172 @@
-  }
-  export interface CustomMenuRendererProps {
-    groupByKey?: string;
-  }
-  export type MenuRendererProps = SelectNS.MenuRendererProps &
-    SelectMissingMenuRendererProps &
-    CustomMenuRendererProps;
-  export type MenuRendererHandler = (
-    props: MenuRendererProps
-  ) => JSX.Element[] | SelectNS.HandlerRendererResult;
-
-  export type MenuPosition = "top" | "bottom";
-
-  export type Props = RequiredProps & Partial<DefaultProps>;
-}
-
-type DefaultedProps = RequiredProps & DefaultProps;
-
 @props(Props, { strict: true })
-export class Dropdown extends React.Component<Dropdown.Props> {
-  static defaultProps: DefaultProps = {
+export class Dropdown<OptionType> extends React.Component<
+  Dropdown.Props<OptionType>
+> {
+  static defaultProps: Dropdown.Props<any> = {
     delimiter: ",",
     size: "medium",
-    disabled: false,
-    searchable: false,
-    clearable: false,
-    multi: false,
+    isDisabled: false,
+    isSearchable: false,
+    isClearable: false,
+    isMulti: false,
     flat: false,
-    autoBlur: true,
-    simpleValue: true,
-    groupByKey: "optionGroup",
-    optionGroupRenderer: defaultOptionGroupRenderer,
-    menuRenderer: defaultMenuRenderer,
+    blurInputOnSelect: true,
     menuPosition: "bottom"
   };
 
-  componentDidMount() {
-    this.logWarnings();
-  }
-
-  logWarnings = () => {
-    if (this.props.children) {
-      warn("You're passing children. Not expected behaviour");
-    }
-  };
-
-  valueToOption = (
-    value: Dropdown.Value | undefined,
-    options: SelectNS.Options
-  ) => {
-    if (t.String.is(value) || t.Number.is(value)) {
-      const { multi, delimiter } = this.props as DefaultedProps;
-      if (multi) {
-        const values = String(value).split(delimiter);
-        return compact(
-          values.map(v => find(options, { value: v }) || undefined)
-        );
-      }
-      return find(options, { value }) || undefined;
-    }
-    return value;
-  };
-
-  sortOptionsByGroup = (options: SelectNS.Options) => {
-    const { groupByKey } = this.props as DefaultedProps;
-    return sortBy(
-      options,
-      option =>
-        option[groupByKey]
-          ? findIndex(options, o => option[groupByKey] === o[groupByKey])
-          : -1
-    );
-  };
-
   getCustomClassNames() {
-    const { size, flat, clearable, menuPosition } = this
-      .props as DefaultedProps;
+    const { size, flat, isClearable, menuPosition, isMulti } = this.props;
     return cx({
       "is-medium": size === "medium",
       "is-small": size === "small",
       "is-flat": flat,
-      "is-clearable": clearable,
+      "is-multi": isMulti,
+      "is-clearable": isClearable,
       "menu-position-top": menuPosition === "top"
     });
   }
 
-  _onChange = (_value?: Dropdown.Value | null) => {
-    const value = _value === "" || isEmptyArray(_value) ? null : _value;
-    return this.props.onChange(value);
-  };
-
-  onInputKeyDown: SelectNS.OnInputKeyDownHandler = e => {
```

Comment:
this was used to manage up and down keys with empty options. I don't know why it was originally implemented but seems useless now..